File: YiAi/src/server/routes/wework.py
"""
WeCom Bot Webhook API
- Provides ability to send messages to WeCom bots
"""
import logging

from fastapi import APIRouter

from domain.wework import send_message
from models.schemas import WeWorkWebhookRequest
from shared.response import success

logger = logging.getLogger(__name__)
router = APIRouter()


@router.post("/wework/send-message", operation_id="send_wework_message")
async def send_wework_message(request: WeWorkWebhookRequest):
    """Send text message to WeCom bot."""
    data = await send_message(request.webhook_url, request.content)
    return success(data=data)

File: YiPet/src/api/services/wework.ts
/**
 * WeCom Bot Webhook API service — sends text messages to a user's
 * configured WeCom bot via the YiAi backend's /wework/send-message route.
 *
 * The backend proxies the request to qyapi.weixin.qq.com and returns
 * {message: "Message sent successfully"} on success.
 */

import type { ApiClient, ApiResponse } from '../client';
import { WEWORK } from '../endpoints';
import type { WeWorkSendMessageParams, WeWorkSendMessageResult } from '../types';

export class WeWorkService {
  constructor(private client: ApiClient) {}

  /** Send a plain text message to a WeCom bot via the configured webhook. */
  async sendMessage(
    params: WeWorkSendMessageParams,
    signal?: AbortSignal,
  ): Promise<ApiResponse<WeWorkSendMessageResult>> {
    return this.client.post<WeWorkSendMessageResult>(
      WEWORK.SEND_MESSAGE,
      {
        webhook_url: params.webhook_url,
        content: params.content,
      },
      signal,
    );
  }
}
